File: plldb-server/src/main/java/cn/twincest/plldb/domain/RawRecord.java
package cn.twincest.plldb.domain;

import com.mongodb.lang.NonNull;
import com.mongodb.lang.Nullable;
import lombok.*;

import java.util.List;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class RawRecord {

	@Nullable
	private String description;

	@NonNull
	private String name;

	@Nullable
	private String author;

	@NonNull
	private Integer sexual;

	@Nullable
	private String source;

	@Nullable
	private Integer determination;

	@NonNull
	private List<Record.Comment> reason;

	@NonNull
	private List<Record.Comment> introduction;

}

File: plldb-server/src-refactor/main/java/cn/twin_kles/plldb/domain/RawRecord.kt
package cn.twin_kles.plldb.domain

class RawRecord(
    description: String?,
    name: String,
    author: String?,
    sexual: Int,
    source: String?,
    determination: Int?,
    reason: List<Record.Comment>,
    introduction: List<Record.Comment>,
) {
    var description: String?
    var name: String
    var author: String?
    var sexual: Int
    var source: String?
    var determination: Int?
    var reason: List<Record.Comment>
    var introduction: List<Record.Comment>

    init {
        this.description = description
        this.name = name
        this.author = author
        this.sexual = sexual
        this.source = source
        this.determination = determination
        this.reason = reason
        this.introduction = introduction
    }
}